### PR TITLE
SDI-643 Model course activity areas 

### PR DESCRIPTION
**Pull request by mikehalmamoj**
As they are created on a Nomis trigger but have a constraint on course activities - so they must be deleted at the same time

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/CourseActivityAreas.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import jakarta.persistence.JoinColumn
+import jakarta.persistence.MapsId
+import jakarta.persistence.OneToOne
+import jakarta.persistence.Table
+import org.hibernate.Hibernate
+import java.io.Serializable
+
+@Entity
+@Table(name = "COURSE_ACTIVITY_AREAS")
+data class CourseActivityAreas(
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/CourseActivityAreas.kt:15]: think we have a convention of using the singular for the class name

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/Repository.kt [hunk 2]
```diff
@@ -244,6 +244,8 @@ class Repository(
     it.courseScheduleRules.size
   }
 
+  fun runInTransaction(block: () -> Unit) = block()
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/Repository.kt:247]: very nice!

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/CourseAreaRepository.kt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.helper
+
+import org.springframework.data.repository.CrudRepository
+import org.springframework.stereotype.Repository
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.CourseActivityAreas
+
+@Repository
+interface CourseAreaRepository : CrudRepository<CourseActivityAreas, Long>
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/CourseAreaRepository.kt:8]: V minor but these repos normally have the full entity name, ie CourseActivityAreaRepository. I know this one is only for test 🤔

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivityResourceIntTest.kt [hunk 4]
```diff
@@ -774,7 +820,14 @@ class ActivityResourceIntTest : IntegrationTestBase() {
...
         .expectBody(CreateActivityResponse::class.java)
         .returnResult().responseBody
-        ?.courseActivityId
+        ?.courseActivityId!!
+
+      // Emulate the Nomis trigger COURSE_ACTIVITIES_T2.trg
+      repository.runInTransaction {
+        val activity = repository.lookupActivity(activityId)
+        activity.area = CourseActivityAreas(activityId, activity, "AREA")
+        repository.activityRepository.save(activity)
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivityResourceIntTest.kt:829]: You dont need the save() do you ?

Comment (mikehalmamoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivityResourceIntTest.kt:829]: It appears that I do... must be related to transactions?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/Repository.kt (hunk 2 above)
Comment (mikehalmamoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/Repository.kt:247]: ©️ @andymarke

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**